File: javascript/utility.js
export const clamp = (value, min, max) => Math.min(Math.max(value, min), max);

export const isEmpty = (value) => value == null || value === '';

export const capitalize = (value) => value.length === 0 ? value : value[0].toUpperCase() + value.slice(1);

export const reverseString = (value) => [...value].reverse().join('');

export const chunk = (items, size) => {
  if (size <= 0) throw new Error('Chunk size must be greater than zero');
  const result = [];
  for (let index = 0; index < items.length; index += size) {
    result.push(items.slice(index, index + size));
  }
  return result;
};

File: python/utility.py
from collections.abc import Iterable


def clamp(value, minimum, maximum):
    return min(max(value, minimum), maximum)


def is_empty(value):
    return value is None or value == ""


def capitalize(value):
    return value[:1].upper() + value[1:] if value else value


def reverse_string(value):
    return value[::-1]


def chunk(items: Iterable, size):
    if size <= 0:
        raise ValueError("Chunk size must be greater than zero")
    items = list(items)
    return [items[index:index + size] for index in range(0, len(items), size)]
